app/models/role.rb

```diff
@@ -149,13 +159,25 @@ def disable_filters_overriding
   end
 
   def clone(role_params = {})
-    new_role = self.deep_clone(:except => [:name, :builtin],
+    new_role = self.deep_clone(:except  => [:name, :builtin, :origin],
                                :include => [:locations, :organizations, { :filters => :permissions }])
     new_role.attributes = role_params
     new_role.cloned_from_id = self.id
     new_role
   end
 
+  def locked?
+    return false unless respond_to? :origin
+    origin.present? && builtin != BUILTIN_DEFAULT_ROLE
+  end
+
+  def ignore_locking(&block)
+    self.modify_locked = true
+    yield self
```

Comment:
I think it'd be better either explicitly call the block by `block.call(self)` or don't specify it's name if you're using `yield`